hudi-common/src/main/java/org/apache/hudi/avro/HoodieAvroUtils.java

```diff
@@ -240,4 +244,27 @@ public static String decompress(byte[] bytes) {
       throw new HoodieIOException("IOException while decompressing text", e);
     }
   }
+
+  /**
+   * Generate a reader schema off the provided writeSchema, to just project out the provided columns.
+   */
+  public static Schema generateProjectionSchema(Schema originalSchema, List<String> fieldNames) {
+    Map<String, Field> schemaFieldsMap = originalSchema.getFields().stream()
+        .map(r -> Pair.of(r.name().toLowerCase(), r)).collect(Collectors.toMap(Pair::getLeft, Pair::getRight));
+    List<Schema.Field> projectedFields = new ArrayList<>();
+    for (String fn : fieldNames) {
+      Schema.Field field = schemaFieldsMap.get(fn.toLowerCase());
```

Comment:
I think we can avoid forming `schemaFieldsMap` here. We can directly get the field from original schema using `originalSchema.getField()` and I don't think there will be any performance impact here because Avro internally maintains a Map to similar to this to find the field instead of `O(n)`